appserver/orb/orb-iiop/src/main/java/org/glassfish/enterprise/iiop/impl/GlassFishORBManager.java

```diff
@@ -952,19 +952,18 @@ private String[] getORBInitRef(String endpoints) {
 
     private String getCorbalocURL(Object[] list) {
 
-        String corbalocURL = "";
+        StringBuilder corbalocURL = new StringBuilder();
         //convert list into corbaloc url
         for (int i = 0; i < list.length; i++) {
             logger.log(Level.INFO, "list[i] ==> {0}", list[i]);
-            if (corbalocURL.equals("")) {
-                corbalocURL = IIOP_URL + ((String) list[i]).trim();
+            if (corbalocURL.toString().equals("")) {
+                corbalocURL.append(IIOP_URL).append(((String) list[i]).trim());
             } else {
-                corbalocURL = corbalocURL + "," +
-                        IIOP_URL + ((String) list[i]).trim();
+                corbalocURL.append(",").append(IIOP_URL).append(((String) list[i]).trim());
             }
         }
-        logger.log(Level.INFO, "corbaloc url ==> {0}", corbalocURL);
-        return corbalocURL;
+        logger.log(Level.INFO, "corbaloc url ==> {0}", corbalocURL.toString());
```

Comment:
Same here. Maybe use Logger.log with supplier.

Reply:
It will be converted to String anyway -- just assign it to a local variable and use both here and in return.
When you touch it -- supplier is really better!
```
String corbalocURLString = corbalocURL.toString();
log.info(() -> "corbaloc url ==> " + corbalocURLString);
```